src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -1012,6 +1023,35 @@ private void ShowUpdatePackages(IVsWindowFrame windowFrame, ShowUpdatePackageOpt
             packageManagerControl?.ShowUpdatePackages(updatePackageOptions);
         }
 
+        public async Task ShowReadMeMarkdownToolWindowAsync(string filePath)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(DisposalToken);
+
+            var toolWindow = FindToolWindow(typeof(ReadMeMarkdownToolWindow), 0, create: true);
+
+            if (!(toolWindow is ReadMeMarkdownToolWindow markdownToolWindow))
```

Comment:
Is it really possible for it to be any other type of window at this point?

Reply:
if it can't the window type, FindToolWindow returns null. line 1032 checks for this before it moves on for safe measure

Reply:
This serves as a null check and cast in one expression so this is fine.

Reply:
> if it can't the window type, FindToolWindow returns null. line 1032 checks for this before it moves on for safe measure

Great, that's what I thought. Then for clarity of code, can you please change to `is null` instead of "not is type" which it's already guaranteed to be?

Reply:
Also, this line makes me believe the `.Frame` could be `null` even when there is a window. Think you need this in your case?
https://github.com/NuGet/NuGet.Client/blob/b9a70ce5618918d1236dc5d26ab0c0377a125951/src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs#L376

Reply:
Bringing back up my original point regarding the cast/null-check...

 It looks to me that it's highly unlikely for `FindToolWindow ` to ever return `null`.

- You're passing `create: true` meaning it will create a window if it doesn't exist.

- The only scenario where a `null` would exist is when `ReadMeMarkdownToolWindow` is created as a **non-Tool window**,  which we aren't doing.

I suggest considering removing the check and `return` altogether as it's just confusing and appears to be doing something that it isn't.

```suggestion
            var toolWindow = (ReadMeMarkdownToolWindow)FindToolWindow(typeof(ReadMeMarkdownToolWindow), 0, create: true);
```